lib/common/Utils.ts

```diff
@@ -1,3 +1,3 @@
 export function isPromise(value: any): value is Promise<any> {
-  return value && value.then !== undefined ? true : false
+  return value instanceof Promise;
```

Comment:
what about PromiseLike entities?

Reply:
Good point. I want to make this line more understandable. How about I change this to - 
```.ts
export function isPromise(value: any): value is Promise<any> {
  return value?.then instanceof Function;
}

Reply:
or this - 
```ts
export function isPromise(value: any): value is Promise<any> {
  return Boolean(value && value.then);
}